src/main/java/org/openstreetmap/osmosis/plugin/elasticsearch/client/ElasticsearchClientBuilder.java

```diff
@@ -91,17 +92,22 @@ protected Client buildTransportClient() {
 		LOG.info(String.format("Connecting to elasticsearch cluster '%s' via [%s]" +
 				" using TransportClient", clusterName, hosts));
 		// Connect as TransportClient (proxy of the cluster)
-		Settings settings = ImmutableSettings.settingsBuilder()
+		Settings settings = Settings.settingsBuilder()
 				.put("cluster.name", clusterName)
 				.put("client.transport.sniff", true)
 				.build();
-		TransportClient transportClient = new TransportClient(settings);
+		TransportClient transportClient = TransportClient.builder().settings(settings).build();
 		// Add specified TransportAddresses
 		for (String host : hosts.split(",")) {
 			String[] params = host.split(":");
 			String hostname = params[0];
 			int port = (params.length == 2) ? Integer.valueOf(params[1]) : 9300;
-			transportClient.addTransportAddress(new InetSocketTransportAddress(hostname, port));
+			try {
+				transportClient.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(hostname), port));
+			} catch (UnknownHostException e) {
```

Comment:
in 2.0+ the InetSocketTransportAddress constructor looks like :
```org.elasticsearch.common.transport.InetSocketTransportAddress.InetSocketTransportAddress(InetAddress address, int port) ```

so it will not automatically cast a String to ```InetAddress```